GocadRelatedScripts/convertTs2Stl.py

```diff
@@ -1,171 +1,31 @@
 import numpy as np
 import argparse
+import sys
+import os
+from Face import Face
 
-parser = argparse.ArgumentParser(description="project ts file")
+
+parser = argparse.ArgumentParser(description="convert ts file to stl")
 parser.add_argument("ts_file", help="ts filename")
-parser.add_argument("--proj", nargs=1, metavar=("projname"), default=(""), help="string describing its projection (ex: +init=EPSG:32646 (UTM46N), or geocent (cartesian global)) if a projection is considered")
-parser.add_argument("--debug", dest="debug", action="store_true", help="print additionnal debug info")
-parser.add_argument("--merged", dest="merged", action="store_true", help="if true, each surface is not isolated in a stl solid")
-parser.add_argument("--bstl", dest="bstl", action="store_true", help="if true, write binary stl instead of stl (note that surface will be merged once imported by SimModeler)")
+parser.add_argument("--proj", nargs=1, metavar=("projname"), default=None, help="string describing its projection (ex: +init=EPSG:32646 (UTM46N), or geocent (cartesian global)) if a projection is considered")
 parser.add_argument("--tokm", dest="tokm", action="store_true", help="convert coordinates to km")
 parser.add_argument("--translate", nargs=2, metavar=("x0", "y0"), default=([0, 0]), help="translates all nodes by (x0,y0)", type=float)
 args = parser.parse_args()
 
 
-# set projection
-if args.proj != "":
-    import mpl_toolkits.basemap.pyproj as pyproj
-
-    lla = pyproj.Proj(proj="latlong", ellps="WGS84", datum="WGS84")
-    if args.proj[0] != "geocent":
-        sProj = args.proj[0]
-        myproj = pyproj.Proj(sProj)
-    else:
-        myproj = pyproj.Proj(proj="geocent", ellps="WGS84", datum="WGS84")
-
-# set logging
-import logging, sys
-
-if args.debug:
-    logging.basicConfig(format="%(message)s", stream=sys.stderr, level=logging.DEBUG)
-else:
-    logging.basicConfig(format="%(message)s", stream=sys.stderr, level=logging.INFO)
-
-surface_name = "undefined"
-sid = 0
-
-if not args.bstl:
-    foutname = args.ts_file[0:-3] + ".stl"
-    fout = open(foutname, "w")
-    if args.merged:
-        fout.write("solid 0\n")
-else:
-    import struct
-
-    foutname = args.ts_file[0:-3] + ".bstl"
-    fout = open(foutname, "wb")
-    fout.seek(80)
-    # number of triangle. will be filled afterwards
-    fout.write(struct.pack("<L", 0))
+idface = 0
+basename, ext = os.path.splitext(args.ts_file)
 
-
-nfacets = 0
 with open(args.ts_file) as fid:
     while True:
```

Comment:
I don't see why we need this `while` any more. myFace will be always the same, right?

Reply:
a ts file can contain several faces. Each of them will be written in a different file (and then not merged into one object by SimModeler)